### PR TITLE
Feature/code improvements

### PR DESCRIPTION
**Pull request by zahardev**

### CODE REVIEW

File: php/classes/controllers/class-settings-controller.php [hunk 1]
```diff
@@ -785,7 +790,8 @@ public function settings_page() {
 			);
 			$html .= '<form method="post" action="' . esc_url_raw( $current_admin_url ) . '" enctype="multipart/form-data">' . "\n";
 			$html .= '<input type="hidden" name="action" value="post_import_form" />';
-			$html .= wp_nonce_field( 'ss_podcasting_import' );
+			$html .= wp_nonce_field( 'ss_podcasting_import', '_wpnonce', true, false );
```
Comment (zahardev) [inline php/classes/controllers/class-settings-controller.php:801]: Fixed a bug - if the 4th parameter is empty, it prints the nonce field, so we had 2 nonce fields from one call - one when it fires, and one when it $html is being printed.

File: php/classes/ajax/class-ajax-handler.php [hunk 2]
```diff
@@ -198,19 +156,13 @@ public function update_episode_embed_code() {
 	 * Import an external RSS feed via ajax
 	 */
 	public function import_external_rss_feed() {
-		// @todo add nonces, add user caps check, validate inputs
+		self::import_security_check();
```
Comment (jonathanbossenger) [inline php/classes/ajax/class-ajax-handler.php]: To stick with the "development style" of the plugin, can this be replaced with 
```
$this->import_security_check();
```

File: php/classes/ajax/class-ajax-handler.php [hunk 3]
```diff
@@ -232,10 +184,60 @@ public function get_external_rss_feed_progress() {
...
+	 * Throws exception if nonce is not valid
+	 *
+	 * @param string $action
+	 * @param string $nonce_key
+	 *
+	 * @throws \Exception
+	 */
+	protected function nonce_check( $action, $nonce_key = 'nonce' ) {
+		if ( ! wp_verify_nonce( $_REQUEST[ $nonce_key ], $action ) ) {
+			throw new \Exception( 'Security error!' );
```
Comment (jonathanbossenger) [inline php/classes/ajax/class-ajax-handler.php:216]: What happens if these exceptions are thrown, does the user see an error message, or does it just fail silently?

File: php/classes/ajax/class-ajax-handler.php (hunk 2 above)
Comment (jonathanbossenger) [reply php/classes/ajax/class-ajax-handler.php]: self:: is usually only needed with static functions.

**Review by jonathanbossenger — CHANGES_REQUESTED** [review]
Looks good, just a few small change requests.

File: php/classes/ajax/class-ajax-handler.php (hunk 2 above)
Comment (zahardev) [reply php/classes/ajax/class-ajax-handler.php]: Ah, good point.

File: php/classes/ajax/class-ajax-handler.php (hunk 3 above)
Comment (zahardev) [reply php/classes/ajax/class-ajax-handler.php:216]: It alerts an error here - https://github.com/CastosHQ/Seriously-Simple-Podcasting/pull/594/files#diff-ce0a39a9329251dfc938d0642fcb5f2eb891e4d796866fa9ce107928bf20506eR104
So, the same behavior as it was before.

**Pull request merged**